pyiceberg/catalog/rest.py

```diff
@@ -532,7 +534,7 @@ def _config_headers(self, session: Session) -> None:
         session.headers["Content-type"] = "application/json"
         session.headers["X-Client-Version"] = ICEBERG_REST_SPEC_VERSION
         session.headers["User-Agent"] = f"PyIceberg/{__version__}"
-        session.headers["X-Iceberg-Access-Delegation"] = "vended-credentials"
+        session.headers["X-Iceberg-Access-Delegation"] = self.properties.get(ACCESS_DELEGATION, ACCESS_DELEGATION_DEFAULT)
```

Comment:
It looks like we already have a way of setting custom headers in `_extract_headers_from_properties`. If I understand it correctly, if we set a property like: `header.X-Iceberg-Access-Delegation = remote-signing` then this should set the header "X-Iceberg-Access-Delegation" as `remote-signing`.

I think we could achieve this by setting the default header values, and then setting the property based values after the default values are set:

```
        session.headers["Content-type"] = "application/json"
        session.headers["X-Client-Version"] = ICEBERG_REST_SPEC_VERSION
        session.headers["User-Agent"] = f"PyIceberg/{__version__}"
        session.headers["X-Iceberg-Access-Delegation"] = "vended-credentials"
        header_properties = self._extract_headers_from_properties()
        session.headers.update(header_properties)
```

What do you think of this approach over introducing a different property?

Reply:
IMHO having an explicit and documented property is better than setting custom header. But I can change if you think it's better.

Reply:
@sungwy I tried your code, but some tests failed with:
```
    assert (
            catalog._session.headers.get("Content-type") == "application/json"
    ), "Expected 'Content-Type' default header not to be overwritten"
```

I guess the idea is not allow default header override. Thus I guess having a dedicate property still the best option. What are you thoughts on this?

Reply:
Hi @guitcastro - I appreciate you giving it a go. That error message is interesting because its specific to the `Content-Type` property, Did you intend to overwrite the `Content-Type` as well?

Reply:
@guitcastro I was able to make @sungwy's suggestion work in https://github.com/guitcastro/iceberg-python/pull/1, without breaking existing tests.